pkg/proxy/winkernel/proxier.go

```diff
@@ -1112,14 +1131,14 @@ func (proxier *Proxier) syncProxyRules() {
 			}
 
 			if len(ep.hnsID) > 0 {
-				newHnsEndpoint, err = hns.getEndpointByID(ep.hnsID)
+				newHnsEndpoint = queriedEndpoints[ep.hnsID]
```

Comment:
Why is this queried here and by ip right after?  This looks valid only in the overlay scenario and only due to previous way this worked. For any endpoints that already exist is would be found by IP.  (similiar to comment above)

Reply:
Updated to only query by IP

Reply:
After more deliberation with @sbangari, I have reverted back to query endpoints by ID first again. Unfortunately it appears there is a bug in Windows Server 2019 that could sometimes cause 2 endpoints to be created with the same IP address, and there is fear of this potentially causing issues if we only query by IP address and remove the call to query by ID.